packages/mui-joy/src/Snackbar/Snackbar.tsx

```diff
@@ -0,0 +1,376 @@
+'use client';
+import * as React from 'react';
+import PropTypes from 'prop-types';
+import clsx from 'clsx';
+import { unstable_composeClasses as composeClasses } from '@mui/base';
+import { ClickAwayListener } from '@mui/base/ClickAwayListener';
+import { useSnackbar } from '@mui/base/useSnackbar';
+import { unstable_capitalize as capitalize } from '@mui/utils';
+import { OverridableComponent } from '@mui/types';
+import { ColorInversionProvider } from '../styles/ColorInversion';
+import useSlot from '../utils/useSlot';
+import styled from '../styles/styled';
+import { useThemeProps } from '../styles';
+import { resolveSxValue } from '../styles/styleUtils';
+import { SnackbarProps, SnackbarOwnerState, SnackbarTypeMap } from './SnackbarProps';
+import { getSnackbarUtilityClass } from './snackbarClasses';
+
+const useUtilityClasses = (ownerState: SnackbarOwnerState) => {
+  const { variant, color, size, anchorOrigin } = ownerState;
+
+  const slots = {
+    root: [
+      'root',
+      size && `size${capitalize(size)}`,
+      color && `color${capitalize(color)}`,
+      variant && `variant${capitalize(variant)}`,
+      `anchorOrigin${capitalize(anchorOrigin!.vertical)}${capitalize(anchorOrigin!.horizontal)}`,
+    ],
+    startDecorator: ['startDecorator'],
+    endDecorator: ['endDecorator'],
+  };
+
+  return composeClasses(slots, getSnackbarUtilityClass, {});
+};
+
+const SnackbarRoot = styled('div', {
+  name: 'JoySnackbar',
+  slot: 'Root',
+  overridesResolver: (props, styles) => styles.root,
+})<{ ownerState: SnackbarOwnerState }>(({ theme, ownerState }) => {
+  const { p, padding, borderRadius } = resolveSxValue({ theme, ownerState }, [
+    'p',
+    'padding',
+    'borderRadius',
+  ]);
+
+  return [
+    {
+      zIndex: theme.vars.zIndex.snackbar,
+      position: 'fixed',
+      display: 'flex',
+      left: 8,
+      right: 8,
+      justifyContent: 'flex-start',
+      alignItems: 'center',
+      minWidth: 300,
+      ...(ownerState.anchorOrigin!.vertical === 'top' ? { top: 8 } : { bottom: 8 }),
+      [theme.breakpoints.up('sm')]: {
```

Comment:
Let's remove media query and do this:

```js
right: `clamp(8px, (${theme.breakpoints.values.sm || '600px'} - 100vw) * 999, 24px)`,
```

with this, when the viewport is larger than `breakpoints.sm`, the value will be '24px' else '8px'.

Reply:
The syntax is wrong, it should be:
```js
left: `clamp(8px, calc(${theme.breakpoints.values.sm || '600'}px - 100vw) * 999, 24px)`
```
---
But when the viewport width is larger than `breakpoints.sm` or `600px`, we want one position property to take `auto`. For example, for a bottom left Snackbar, it should be like this:

- On screens larger than `600px`:
  ```js
  bottom: left: `clamp(8px, calc(${theme.breakpoints.values.sm || '600'}px - 100vw) * 999, 24px)`
  left: `clamp(8px, calc(${theme.breakpoints.values.sm || '600'}px - 100vw) * 999, 24px)`,
  right: 'auto',
  ```

- On screens smaller than or equal to `600px`:
  ```js
  bottom: `clamp(8px, calc(${theme.breakpoints.values.sm || '600'}px - 100vw) * 999, 24px)`
  left: `clamp(8px, calc(${theme.breakpoints.values.sm || '600'}px - 100vw) * 999, 24px)`,
  right: `clamp(8px, calc(${theme.breakpoints.values.sm || '600'}px - 100vw) * 999, 24px)`,
  ```

This behavior is consistent with Material UI, unless we want it to be different. How to do that?